pkg/build/api/v1/types.go

```diff
@@ -658,6 +734,27 @@ type BuildRequest struct {
 
 	// Env contains additional environment variables you want to pass into a builder container
 	Env []kapi.EnvVar `json:"env,omitempty"`
+
+	// BuildCauses can pass information from the controllers to the buildconfig
+	// generator about who/what has triggered a build through the request.
+	BuildCauses []*BuildTriggerCauses `json:"BuildCauses,omitempty"`
```

Comment:
I'm not up-to-date with the discussions here, but if you decide to move away from a slice, just keep in mind that deployments use a [slice of causes](https://github.com/openshift/origin/blob/2ddfc71bf4f24d35afbf4085b1e26ae64bffa107/pkg/deploy/api/v1/types.go#L323).